### PR TITLE
Update forms documentation

### PR DESCRIPTION
**Pull request by himdel**
We now have 2 ui docs about forms:

`forms` - documents pre-react state which is not useful => a copy exists in the [wiki](https://github.com/ManageIQ/manageiq-ui-classic/wiki/Forms-%28kinds%29), removing from guides.

`react-forms` - documents pre-data driven forms state, long obsolete, removed in https://github.com/ManageIQ/react-ui-components/pull/179 .

That's encouraging potential developers to use obsolete methods, we need to remove those.

Data driven forms already have [documentation](http://data-driven-forms.surge.sh), so mostly making the new document link to that + a plugin example (ManageIQ/manageiq-providers-redfish#107).

Cc @skateman

### CODE REVIEW

File: ui/forms.md [hunk 1]
```diff
@@ -1,256 +1,13 @@
 ## Forms
 
-This attempts to document the state of forms in ui-classic, including all the stages of conversion, from pure rails to Angular.
+The recommended way of writing ManageIQ UI forms is using [Data Driven Forms](http://data-driven-forms.surge.sh/).
```
Comment (skateman) [inline ui/forms.md:3]: https://data-driven-forms.org/

File: ui/forms.md [hunk 2]
```diff
@@ -1,256 +1,13 @@
...
-    * queries the server for the data (if editing)
-      * later this will be done in the router
-    * keeps all the form state in it's model object
-    * does all the validation that makes sense
-    * submits that object on save
-    * doesn't talk to the server unless it has to
-  * replace the original view with the use of such a component, with any necessary options coming from server-side
-  * handle only the initial rendering, data, and submit/cancel ruby side
-
+For data driven form specific details, please consult [ddf documentation](http://data-driven-forms.surge.sh/).
```
Comment (skateman) [inline ui/forms.md:11]: https://data-driven-forms.org/

File: ui/forms.md [hunk 3]
```diff
@@ -1,256 +1,13 @@
...
-  * the main characteristics of such a form with pointers where to look
-  * the list of actions involved during the form's lifecycle
-  * a code example of a minimal form with
-    * one text input validated for length
-    * one select, the choice in which updates a useless message div
-    * the default form buttons
-  * a recommended migration path
+---
 
+Manageiq UI imports data driven forms and exposes a `@@ddf` webpack alias for both local and plugin imports.
```
Comment (skateman) [inline ui/forms.md:9]: Would be nice to have a little bit more, in a form of example maybe:
```
import MiqFormRenderer, { componentTypes, validatorTypes } from '@@ddf';
```

The `MiqFormRenderer` is our wrapper around DDF with our custom components and implicitly translated button labels. For more information see the definition of the component (should have a link to data-driven-form.jsx). The `componentTypes` and `validatorTypes` are a set of constants provided by DDF for creating a schema.

**Comment by himdel** [comment]
Added a minimalistic example :)

File: ui/forms.md [hunk 4]
```diff
@@ -1,256 +1,82 @@
...
```
Comment (chessbyte) [inline ui/forms.md:9]: Typo: `Manageiq` should be `ManageIQ`

Comment (himdel) [reply ui/forms.md]: Fixed :)

**Pull request merged**